src/sources/splunk_hec/mod.rs

```diff
@@ -267,22 +267,41 @@ impl SplunkSource {
                             }
                             _ => None,
                         };
-                        let mut events = stream::iter(EventIterator::new(
+
+                        let mut error = None;
+                        let events = EventIterator::new(
                             Deserializer::from_str(&body).into_iter::<JsonValue>(),
                             channel,
                             remote,
                             xff,
                             batch,
                             token.filter(|_| store_hec_token).map(Into::into),
-                        ));
+                        )
+                        .map_while(|result| match result {
+                            Ok(event) => Some(event),
+                            Err(err) => {
+                                error = Some(err);
+                                None
+                            }
+                        })
```

Comment:
Nit: I've never been a big fan of having a mutable `error` like this in a closure. I think I'd prefer a `for` loop 🤔

Reply:
Yeah, looking again I think I agree. It's not quite as concise but easier to understand.